orchestra/communication/staffing.py

```diff
@@ -122,48 +127,130 @@ def check_responses_complete(request):
              .format(request.task)))
 
 
-def send_staffing_requests(
+def address_staffing_requests(
         worker_batch_size=settings.ORCHESTRA_STAFFBOT_WORKER_BATCH_SIZE,
         frequency=settings.ORCHESTRA_STAFFBOT_BATCH_FREQUENCY):
     staffbot = StaffBot()
     cutoff_datetime = timezone.now() - frequency
     requests = (
         StaffBotRequest.objects
-        .filter(status=StaffBotRequest.Status.SENDING_INQUIRIES.value)
+        .filter(status__in=[
+            StaffBotRequest.Status.SENDING_INQUIRIES.value,
+            StaffBotRequest.Status.DONE_SENDING_INQUIRIES.value])
         .filter(Q(last_inquiry_sent__isnull=True) |
-                Q(last_inquiry_sent__lte=cutoff_datetime)))
+                Q(last_inquiry_sent__lte=cutoff_datetime))
+        .order_by('-task__project__priority', 'created_at'))
 
     for request in requests:
-        send_request_inquiries(staffbot, request, worker_batch_size)
+        staff_or_send_request_inquiries(staffbot, request, worker_batch_size)
+
+
+def _is_worker_assignable(worker, task, required_role):
+    try:
+        check_worker_allowed_new_assignment(worker)
+        if (is_worker_certified_for_task(worker, task,
+                                         required_role,
+                                         require_staffbot_enabled=True)
+                and not task.is_worker_assigned(worker)):
+            return True
+    except TaskStatusError:
+        pass
+    except TaskAssignmentError:
+        pass
+    return False
+
+
+def _can_handle_more_work_today(worker, task):
+    can_handle_more_hours = False
+    today = timezone.now().date()
+    today_abbreviation = ['mon', 'tues', 'wed', 'thurs', 'fri', 'sat', 'sun'][
+        today.weekday()]
+    availability = WorkerAvailability.objects.filter(
+        worker=worker,
+        week=first_day_of_the_week()).first()
+    task_hours = task.get_assignable_hours()
+    if availability is not None and task_hours is not None:
+        desired_hours = getattr(
+            availability, 'hours_available_{}'.format(today_abbreviation), 0)
+        allowed_hours = min(desired_hours, worker.max_autostaff_hours_per_day)
+        responses = StaffingResponse.objects.filter(
```

Comment:
add a comment (so we don't have to rediscover later) that this only includes tasks you won via StaffBot, meaning you can be manually assigned work and then still get StaffBotted even more

Reply:
Because we're looking at StaffingResponse objects to determine assigned tasks for a worker, we don't consider tasks that were assigned manually (without an open staffbot request). For example, we ignore tasks staffed by directly typing a worker's username into the projman interface when we had never previously opened a staffbot request for the task.